vignettes/column-types.Rmd

```diff
@@ -88,77 +47,30 @@ tfile <- tempfile("tricky-column-type-guessing-", fileext = ".csv")
 write_csv(tricky_dat, tfile)
 ```
 
-First, note that the second edition parser guesses the right type for `x`, even with the default `guess_max` behaviour.
+The first edition parser doesn't guess the right type for `x` so the `2` becomes an `NA`:
 
 ```{r}
-tail(read_csv(tfile))
+df <- with_edition(1, read_csv(tfile))
+tail(df)
 ```
 
-In contrast, the first edition parser doesn't guess the right type for `x` with the `guess_max` default.
-`x` is imported as logical and the `2` becomes an `NA`.
+For this specific case, we can fix the problem by marginally increasing `guess_max`:
 
 ```{r}
-with_edition(
-  1,
-  tail(read_csv(tfile))
-)
+df <- with_edition(1, read_csv(tfile, guess_max = 1001))
+tail(df)
+```
+
+Unlike the 2nd edition, we don't recommend using `guess_max = Inf` because the engine pre-allocates a large amount of memory in the face of this uncertainty.
```

Comment:
```suggestion
Unlike the 2nd edition, we don't recommend using `guess_max = Inf` with the legacy parser, because the engine pre-allocates a large amount of memory in the face of this uncertainty.
```